Libraries/JavaScriptAppEngine/System/JSTimers/JSTimersExecution.js

```diff
@@ -70,6 +73,12 @@ var JSTimersExecution = {
       } else if (type === JSTimersExecution.Type.requestAnimationFrame) {
         var currentTime = performanceNow();
         callback(currentTime);
+      } else if (type === JSTimersExecution.Type.requestIdleCallback) {
+        callback({
+          timeRemaining: function() {
+            return Math.max(0, 17 - (performanceNow() - frameTime));
```

Comment:
I don't think the debugging argument is significant here because this whole API is a massive race condition regardless. There's no reliable break debugging strategy here anyway. A TODO would be good though.